subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.groovy

```diff
@@ -161,7 +161,9 @@ class Checkstyle extends SourceTask implements VerificationTask, Reporting<Check
             }
 
             if (reports.html.enabled) {
-                def xsl = Checkstyle.getClassLoader().getResourceAsStream('checkstyle-noframes-sorted.xsl')
+                def xsl = reports.html.stylesheet ?
+                        new FileInputStream(reports.html.stylesheet as String) :
```

Comment:
It is. Yet it also depends whether the Checkstyle and Findbugs plugins should be consistent regarding where the XSLT stylesheet can be loaded from.

For the Checkstyle plugin, the XSLT transformation is performed in-process using the [Ant XSLT task](https://github.com/gradle/gradle/blob/REL_2.11/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.groovy#L165), hence the XSLT stylesheet can be in the same classpath.

For the FindBugs plugin, the transformation is performed out-of-process by Findbugs itself (in the [process spawned](https://github.com/gradle/gradle/blob/REL_2.11/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugs.groovy#L257) by the Gradle task). Hence the transformation executes in a separate classpath.